client_integration_test.go

```diff
@@ -1249,55 +1249,50 @@ func TestClientReadSequential(t *testing.T) {
 	}
 }
 
+// this writer requires maxPacket = 3 and always returns an error for the second write call
 type lastChunkErrSequentialWriter struct {
-	expected      int
-	written       int
-	writtenReturn int
+	writeCounter int
 }
 
 func (w *lastChunkErrSequentialWriter) Write(b []byte) (int, error) {
-	chunkSize := len(b)
-	w.written += chunkSize
-	if w.written == w.expected {
-		return w.writtenReturn, errors.New("test error")
+	if len(b) > 3 {
+		return 0, errors.New("this writer require maxPacket <= 3, please set MaxPacketChecked(2)")
 	}
-	return chunkSize, nil
+	w.writeCounter++
+	if w.writeCounter == 1 {
+		return len(b), nil
+	}
+	return 1, errors.New("this writer fails after the first write")
 }
 
-func TestClientWriteSequential_WriterErr(t *testing.T) {
-	sftp, cmd := testClient(t, READONLY, NODELAY)
+func TestClientWriteSequentialWriterErr(t *testing.T) {
+	client, cmd := testClient(t, READONLY, NODELAY, MaxPacketChecked(3))
 	defer cmd.Wait()
-	defer sftp.Close()
+	defer client.Close()
 
 	d, err := ioutil.TempDir("", "sftptest-writesequential-writeerr")
 	require.NoError(t, err)
 
 	defer os.RemoveAll(d)
 
-	var (
-		content    = []byte("hello world")
-		shortWrite = 2
-	)
-	w := lastChunkErrSequentialWriter{
-		expected:      len(content),
-		writtenReturn: shortWrite,
-	}
-
 	f, err := ioutil.TempFile(d, "write-sequential-writeerr-test")
 	require.NoError(t, err)
 	fname := f.Name()
-	n, err := f.Write(content)
+	_, err = f.Write([]byte("12345"))
 	require.NoError(t, err)
-	require.Equal(t, n, len(content))
 	require.NoError(t, f.Close())
 
-	sftpFile, err := sftp.Open(fname)
+	sftpFile, err := client.Open(fname)
 	require.NoError(t, err)
 	defer sftpFile.Close()
 
-	gotWritten, gotErr := sftpFile.writeToSequential(&w)
-	require.NotErrorIs(t, io.EOF, gotErr)
-	require.Equal(t, int64(shortWrite), gotWritten)
+	w := &lastChunkErrSequentialWriter{}
+	written, err := sftpFile.writeToSequential(w)
+	assert.Error(t, err)
+	if written != 4 {
+		t.Errorf("sftpFile.Write() = %d, but expected 4", written)
+	}
+	assert.Equal(t, 2, w.writeCounter)
```

Comment:
[nit] This test is relying on the default typing of untyped integers assigned to `interface{}` to be `int`, and thus not as stable as it could be.